portfolio/src/main/webapp/script.js

```diff
@@ -12,26 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Identity theft is not a joke', 'I declare bankruptcy', 'I am a little stitious'];
+getComments();
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+// submit a comment
+document.querySelector('.comment-form').addEventListener('submit', async function(e) {
+  e.preventDefault();
+  const response = await fetch('/comment', {
+    method: 'POST',
+    body: new URLSearchParams(new FormData(event.target)),
+  });
+  const json = await response.json();
+  if ('error' in json) {
+    console.log('Could not submit comment');
+  }
+  const commentContainer = document.querySelector('.comments');
+  commentContainer.innerHTML += '<div class="comment">' + document.querySelector('.comment-box').value + '</div><br>';
+});
 
 async function getComments() {
-  const response = await fetch('/data');
+  const response = await fetch('/comment');
   const json = await response.json();
-  const commentContainer = document.getElementById('comments');
+  const commentContainer = document.querySelector('.comments');
   for (const comment of json.comments) {
-    commentContainer.innerHTML += comment;
+    commentContainer.innerHTML += '<div class="comment">' + comment + '</div><br>';
```

Comment:
Unfortunately this is a somewhat dangerous security pattern, as `comment` might have special characters inside (`<`, `"`, etc). Instead, you should use the `.innerText` property to safely set the value.

Reply:
Changed to using createElement() along with .innerText to be safer.